ggml/src/ggml-sycl.cpp

```diff
@@ -5162,13 +5010,19 @@ bool ggml_sycl_compute_forward(ggml_backend_sycl_context & ctx, struct ggml_tens
     return true;
 }
 
+GGML_API GGML_CALL int ggml_backend_sycl_get_device_id(int index) {
```

Comment:
there should never be 2 kinds of index again.

Reply:
The default_tensor_split[] is accessed one by one by device index, instead of device ID.
So, we need a function to convert  from index to ID, or ID to index.

Reply:
why there must be device ID and index again? either ```ONEAPI_DEVICE_FILTER``` or ```CUDA_VISIABLE_DEVICE``` is set, users can only see one kind of index

Reply:
The explain as above comment.
Keep original device ID be stable, will help give clear device info during usage and development.

Reply:
Addtinoal, this API provide the C style interface to ggml lib which is C style.
It would be useful to support more caller, instead of only llama.cpp.

I suggest keeping this API.